src/legacy/core_plugins/console/public/quarantined/src/autocomplete/components/url_pattern_matcher.js

```diff
@@ -54,6 +56,9 @@ export class UrlPatternMatcher {
     endpoint.methods.forEach((method) => {
       let c;
       let activeComponent = this[method].rootComponent;
+      if (endpoint.template) {
+        new FullRequestComponent(pattern + '[body]', activeComponent, endpoint.template);
```

Comment:
I might be missing something, but it doesn't look like we're assigning this to a variable, so what is this doing? I read through the constructors in the inheritance chain and didn't see any notable side effects.

Reply:
This function call results in assignment in the component hierarchy:

https://github.com/elastic/kibana/blob/0eb4c18fe09c3f01bee3ec16206738a3a53f78ae/src/legacy/core_plugins/console/public/np_ready/lib/autocomplete/components/shared_component.js#L26

`SharedComponent` is the common ancestor to components like `FullRequestComponent`. It's definitely not super clear at the moment. Perhaps a readme for how the autocomplete system works could be good? Wdyt?

Reply:
Ahh, I see!

> Perhaps a readme for how the autocomplete system works could be good?

This would certainly clear up a lot for me, but if it's time-consuming then I'm not sure it'd be worth it. Maybe we could begin with a brief outline of the roles of each component in the component hierarchy?